hedera-node/src/main/java/com/hedera/services/ledger/accounts/staking/StakeChangeManager.java

```diff
@@ -0,0 +1,181 @@
+import static com.hedera.services.ledger.properties.AccountProperty.DECLINE_REWARD;
+import static com.hedera.services.ledger.properties.AccountProperty.STAKED_ID;
+import static com.hedera.services.ledger.properties.AccountProperty.STAKED_TO_ME;
+
+@Singleton
+public class StakeChangeManager {
+	private final StakeInfoManager stakeInfoManager;
+	private final Supplier<MerkleMap<EntityNum, MerkleAccount>> accounts;
+
+	@Inject
+	public StakeChangeManager(final StakeInfoManager stakeInfoManager,
+			final Supplier<MerkleMap<EntityNum, MerkleAccount>> accounts) {
+		this.stakeInfoManager = stakeInfoManager;
+		this.accounts = accounts;
+	}
+
+	public void withdrawStake(final long curNodeId, final long amount, final boolean declinedReward) {
+		final var node = stakeInfoManager.mutableStakeInfoFor(curNodeId);
+		node.removeRewardStake(amount, declinedReward);
+	}
+
+	public void awardStake(final long newNodeId, final long amount, final boolean declinedReward) {
+		final var node = stakeInfoManager.mutableStakeInfoFor(newNodeId);
+		node.addRewardStake(amount, declinedReward);
+	}
+
+	public long getAccountStakeeNum(final Map<AccountProperty, Object> changes) {
+		final var entityId = (long) changes.getOrDefault(STAKED_ID, 0L);
+		// Node ids are negative
+		return (entityId < 0) ? 0 : entityId;
+	}
+
+	public long getNodeStakeeNum(final Map<AccountProperty, Object> changes) {
+		final var entityId = (long) changes.getOrDefault(STAKED_ID, 0L);
+		// Node ids are negative
+		return (entityId < 0) ? entityId : 0;
+	}
+
+	public static long finalBalanceGiven(
+			@Nullable final MerkleAccount account,
+			@NotNull final Map<AccountProperty, Object> changes
+	) {
+		if (changes.containsKey(BALANCE)) {
+			return (long) changes.get(BALANCE);
+		} else {
+			return (account == null) ? 0 : account.getBalance();
+		}
+	}
+
+	public boolean finalDeclineRewardGiven(
+			@Nullable final MerkleAccount account,
+			@NotNull final Map<AccountProperty, Object> changes
+	) {
+		if (changes.containsKey(DECLINE_REWARD)) {
+			return (Boolean) changes.get(DECLINE_REWARD);
+		} else {
+			return account != null && account.isDeclinedReward();
+		}
+	}
+
+	public long finalStakedToMeGiven(
+			@Nullable final MerkleAccount account,
+			@NotNull final Map<AccountProperty, Object> changes
+	) {
+		if (changes.containsKey(STAKED_TO_ME)) {
+			return (long) changes.get(STAKED_TO_ME);
+		} else {
+			return (account == null) ? 0 : account.getStakedToMe();
+		}
+	}
+
+	public void updateStakedToMe(
+			@NotNull final int stakeeI,
+			final long delta,
+			@NotNull final EntityChangeSet<AccountID, MerkleAccount, AccountProperty> pendingChanges
+	) {
+		final var mutableChanges = pendingChanges.changes(stakeeI);
+		if (mutableChanges.containsKey(STAKED_TO_ME)) {
+			mutableChanges.put(STAKED_TO_ME, (long) mutableChanges.get(STAKED_TO_ME) + delta);
+		} else {
+			mutableChanges.put(STAKED_TO_ME, delta);
```

Comment:
⚠️ When committed, this will set the **entire** `stakedToMe` balance for this account to `delta`. We need to use,
```
final var newStakedToMe = pendingChanges.entity(stakeeI). getStakedToMe() + delta`;
mutableChanges.put(STAKED_TO_ME, newStakedToMe);
```